### PR TITLE
[DataGrid] Improve edit cell UI

### PR DESCRIPTION
**Pull request by oliviertassinari**
A follow-up on #1025. Proposed design:

Screenshot 2021-03-05 at 00 26 50

I have more or less reproduced the outlined style.

### CODE REVIEW

**Comment by oliviertassinari** [comment]
I have added a new section in the storybook for dedicated visual regressions tests. I think that we could progressively remove the visual regressions on the other stories. If these other stories are for testing edge cases, or stress tests, they probably don't need to be visually tested. If they are important, then they should either have a demo in the documentation or have a story under the visual regression folder. This would reproduce the tradeoff we have in the main repository. The only difference is that we have no tools to easily display the visual regressions.

File: packages/storybook/src/stories/regressions-grid.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { XGrid, useGridApiRef } from '@material-ui/x-grid';
+
+export default {
+  title: 'regressions/Grid',
```
Comment (dtassone) [inline packages/storybook/src/stories/regressions-grid.stories.tsx:5]: Not sure about this approach.
seems a bit segregated and I prefer to test some stories within their thematic folders. Maybe a prefix on the regressions tested stories would allow us to identify which one are tested?

Comment (oliviertassinari) [reply packages/storybook/src/stories/regressions-grid.stories.tsx:5]: I could do a prefix. A bit harder to find and filter but they would be, as you said, in a thematic folder.

**Comment by DanailH** [comment]
I'm ok with the design, it does seem more intuitive.

File: packages/storybook/src/stories/regressions-grid.stories.tsx (hunk 1 above)
Comment (oliviertassinari) [reply packages/storybook/src/stories/regressions-grid.stories.tsx:5]: I guess that also the value of not using a semantic folder. When you want to add a new regression test case, you don't need to care about it.

Comment (oliviertassinari) [reply packages/storybook/src/stories/regressions-grid.stories.tsx:5]: For instance, here, now I need to ask: where should it be?

Comment (oliviertassinari) [reply packages/storybook/src/stories/regressions-grid.stories.tsx:5]: I'm trying something, we will see

File: packages/storybook/src/stories/grid-rows.stories.tsx [hunk 2]
```diff
@@ -629,3 +629,18 @@ export function CommodityEdit() {
     </React.Fragment>
   );
 }
+
+export function RegressionEditCell() {
```
Comment (dtassone) [inline packages/storybook/src/stories/grid-rows.stories.tsx:633]: it could also be a suffix as well

- EditCellSnap?

Note: It seems more like a snapshot test 🤔

File: packages/storybook/src/stories/regressions-grid.stories.tsx (hunk 1 above)
Comment (dtassone) [reply packages/storybook/src/stories/regressions-grid.stories.tsx]: image

Somewhere around here...
ie Commodity Edit Cell Snap ?
Suffix might actually be better 🤔
